src/main/java/mServer/crawler/sender/MediathekArd.java

```diff
@@ -337,125 +326,139 @@ private void filmSuchen2(String urlFilm_, String thema, String titel, long dauer
-                seite2.extractList("_quality\":\"auto\",\"_server\":\"\",\"_cdn\":\"flashls\",\"_stream\":\"", "\"", liste);
-                seite2.extractList("\"_quality\":2", "\"_stream\":[\"", "\"", liste);
-                for (String s : liste) {
-                    if (s.startsWith("http")) {
-                        if (url.isEmpty()) {
-                            url = s;
-                        } else {
-                            urlMid = s;
-                        }
-                        break;
-                    }
-                }
-                liste.clear();
-                seite2.extractList("\"_quality\":1", "\"_stream\":\"", "\"", liste);
-                seite2.extractList("\"_quality\":1", "\"_stream\":[\"", "\"", liste);
-                for (String s : liste) {
-                    if (s.startsWith("http")) {
-                        urlKl = s;
-                        break;
-                    }
-                }
-                if (!urlTest.isEmpty() && (urlTest.equals(url) || urlTest.equals(urlKl))) {
-                    urlHD = ""; //dann ists kein HD
-//                    System.out.println("q3 test: " + urlFilm);
-                }
-//                if (!urlTest.isEmpty() && !(urlTest.equals(url) || urlTest.equals(urlMid))) {
-//                    System.out.println("q3 test: " + urlFilm);
-//                }
-//                if (url.isEmpty() && urlMid.isEmpty() && urlKl.isEmpty() && !thema.equals("alpha-Centauri")) {
-//                    System.out.println("q3 test: " + urlFilm);
-//                }
-//                if (urlMid.isEmpty()) {
-//                    System.out.println("q2test: " + urlFilm);
-//                }
-//                if (urlKl.isEmpty()) {
-//                    System.out.println("q1 test: " + urlFilm);
-//                }
-                if (url.isEmpty()) {
-                    url = urlMid;
-                    urlMid = "";
-                }
-                if (url.isEmpty()) {
-                    url = urlKl;
-                    urlKl = "";
-                }
-                if (url.isEmpty() && !urlHD.isEmpty()) {
-                    url = urlHD;
-                    urlHD = "";
-                }
-                if (urlKl.isEmpty()) {
-                    urlKl = urlMid;
-                }
-                String subtitle = seite2.extract("subtitleUrl\":\"", "\"");
-                if (!subtitle.isEmpty()) {
-                    if (!subtitle.startsWith("http://www.ardmediathek.de")) {
-                        subtitle = "http://www.ardmediathek.de" + subtitle;
-                    }
-                }
-                if (!url.isEmpty()) {
+				seite2.extractList("\"_quality\":2", "\"_stream\":\"", "\"", liste);
+				seite2.extractList("\"_quality\":2", "\"_stream\":[\"", "\"", liste);
+				for (String s : liste) {
+					if (s.startsWith("http")) {
+						if (url.isEmpty()) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)